src/components/portfolio/DetailsMainSummary.tsx

```diff
@@ -0,0 +1,235 @@
+import ValueOverName from './ValueOverName';
+import { Position, RiskAnalysis } from './DetailsTypes';
+import DetailsDonut from './DetailsDonut';
+
+// stylesheet for the Summary section
+const useStyles = makeStyles(({ palette }: Theme) =>
+  createStyles({
+    titleContainer: {
+      display: 'flex',
+      marginBottom: '2rem',
+    },
+    titleWrapper: {
+      marginRight: '1rem',
+    },
+    sectionTitle: {
+      margin: 0,
+      color: palette.primary.contrastText,
+      // TODO use theme fontsize and weight
+      fontSize: '2.25rem',
+      fontWeight: 400,
+      whiteSpace: 'nowrap',
+    },
+    lineWrapper: {
+      display: 'flex',
+      width: '100%',
+      // TODO: use theme color
+      borderColor: 'grey',
+    },
+    line: {
+      width: '100%',
+      alignSelf: 'center',
+      paddingLeft: '2%',
+    },
+    infoBox: {
+      outlineStyle: 'solid',
+      outlineColor: 'grey',
+      outlineWidth: '0.15rem',
+      margin: '1rem 0',
+    },
+    infoValueContainer: {
+      display: 'flex',
+      justifyContent: 'space-evenly',
+    },
+    infoValueWrapper: {
+      display: 'flex',
+      justifyContent: 'space-around',
+      width: '100%',
+    },
+    vl: {
+      margin: '0.5rem 1rem',
+      height: '4rem',
+      alignSelf: 'center',
+      // TODO: use theme color
+      borderColor: 'grey',
+    },
+    chartContainer: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      marginTop: '2rem',
+    },
+    pieChartWrapper: {
+      display: 'flex',
+      width: '20rem',
+      height: '20rem',
+      flexBasis: '35%',
+    },
+    lineChartWrapper: {
+      width: '40rem',
+      height: '20rem',
+      flexBasis: '60%',
+    },
+  })
+);
+
+// type declarations
+type DetailsMainSummaryProps = {
+  // total score of the portfolio
+  score: number;
+  // seven day moving average of the portfolio
+  perf7d: number;
+  // one year moving average of the portfolio
+  perf1y: number;
+  // total value of the portfolio
+  value: number;
+  // number of positions
+  positionCount: number;
+  // all risk fields (country, industry, currency)
+  risk: RiskAnalysis;
+  // array with all positions
+  positions: Position[];
+};
+
+// returns the details page header
+const DetailsMainSummary: React.FC<DetailsMainSummaryProps> = ({
+  score,
+  perf7d,
+  perf1y,
+  value,
+  positionCount,
+  risk,
+  positions,
+}) => {
+  const classes = useStyles();
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  // TODO: no hard coded colors
+  // takes a percent value and converts it to a color
+  function convertPercentToColor(val: number): string {
+    return val < 0 ? '#D64745' : '#50E2A8';
+  }
+
+  // TODO: no hard coded colors
+  // TODO: update range to fit data from analytics
+  // convert a score to a color
+  function convertScoreToColor(val: number): string {
+    return val < 0.5 ? '#D64745' : '#50E2A8';
+  }
+
+  // these are for the DetailsDonut Chart for the company portions
```

Comment:
Why isn't this calculated inside the `Details Donut` component?

Reply:
@an-p Some Details Donut take company names and monetary share of the total value, some take countries and an amount in int... you can't just pass the positions as a prop and calculate it inside DetailsDonut.
See DetailsMainRisk for another usage